dev/WebView2/WebView2.h

```diff
@@ -0,0 +1,286 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+
+#include "pch.h"
+#include "common.h"
+
+#include <winrt\Microsoft.Web.WebView2.Core.h>
+
+namespace winrt
+{
+    using namespace Microsoft::Web::WebView2::Core;
+}
+
+#include "CoreWebView2InitializedEventArgs.g.h"
+
+#include "WebView2.g.h"
+#include "WebView2.properties.h"
+#pragma warning( push )
+#pragma warning( disable : 28251 26812)   // warning C28251: Inconsistent annotation for function: this instance has an error
+#include <webview2.h>
+#include <WebView2EnvironmentOptions.h>
+#pragma warning( pop ) 
+
+ // for making async/await possible
+struct AsyncWebViewOperations final : public Awaitable
+{
+    AsyncWebViewOperations() = default;
+
+    void StopWaiting()
+    {
+        // async operation completed, stop all awaits
+        CompleteAwaits();
+    }
+};
+
+class CoreWebView2InitializedEventArgs :
+    public winrt::implementation::CoreWebView2InitializedEventArgsT<CoreWebView2InitializedEventArgs>
+{
+public:
+    CoreWebView2InitializedEventArgs(winrt::hresult Exception) : m_exception(Exception) {};
+
+    void Exception(winrt::hresult exception) { m_exception = exception; };
+    winrt::hresult Exception() { return m_exception; }
+private:
+    winrt::hresult m_exception{};
+};
+
+template <typename D, typename T, typename ... I>
+struct __declspec(empty_bases) DeriveFromContentControlHelper_base : winrt::Windows::UI::Xaml::Controls::ContentControlT<D, winrt::default_interface<T>, winrt::composable, I...>
+{
+    using composable = T;
+    using class_type = typename T;
+
+    operator class_type() const noexcept
+    {
+        return static_cast<winrt::IInspectable>(*this).as<class_type>();
+    }
+
+    hstring GetRuntimeClassName() const
+    {
+        return hstring{ winrt::name_of<T>() };
+    }
+};
+
+class WebView2 :
+    public ReferenceTracker<WebView2, DeriveFromContentControlHelper_base, winrt::WebView2>,
```

Comment:
>DeriveFromContentControlHelper_base [](http://example.com/codeflow?start=38&length=35)

Why is WV2 deriving from ContentControl (not Control)? I understand that as a Control, WebView2 maps onto the ContentControl concept, but I don't think we should think of WebView2 as Control, but rather a plain UIElement (like legacy WebView) The reason is, with general retemplating, Focus / keyboard navigation paths (which are highly customized to interact with the CWV2) get broken and are quite difficult to repair; this may impact accessibility for the entire app.

It's true that in OS Xaml we need Control for proper Focus API support, so for now I would recommend deriving from UserControl (which specifcally ignores the Template property).

Reply:
ContentControl because that lets us set a child. Control doesn't let you set a child.

Reply:
I haz confus. What is the child of a wv2? i thought we were just putting the control there and pointing it at an url/string to render?

Reply:
Correct, and that's done with the Composition APIs. But when the WV2 runtime can't be found we need to display a TextBlock with hyperlink saying we couldn't find the runtime and telling the user how to go acquire the runtime.